app/context/authContext.tsx

```diff
@@ -0,0 +1,204 @@
+'use client';
+
+import React, { createContext, useContext, ReactNode, useMemo, useEffect } from 'react';
+import { useRouter } from 'next/navigation';
+import { MutationStatus, QueryStatus, useMutation, useQuery } from '@tanstack/react-query';
+import { clearSession, getSession, setSession } from '../../utils/actions/session';
+import getAuthUrl from '../../utils/getAuthUrl';
+
+const baseUrl = process.env.NEXT_PUBLIC_API_BASE_URL || '';
+
+export type Role = 'admin' | 'content_manager' | 'member';
+
+export type MiddlewareType = 'auth' | 'guest';
+
+type UserType = {
+  id: number | string;
+  is_verified: boolean;
+  email: string;
+  role: Role;
+};
+
+interface AuthContextType {
+  user?: UserType | null;
+  login: ({
+    // eslint-disable-next-line no-unused-vars
+    email, // eslint-disable-next-line no-unused-vars
+    password, // eslint-disable-next-line no-unused-vars
+    role,
```

Comment:
can we create an interface here? also for the return type, so if it somehow changes in the future we don't have to remember and make the changes everywhere in the code, we would just change the interface